### PR TITLE
Add ignore for build folder

### PR DESCRIPTION
**Pull request by salimoneus**
### This pullrequest changes

Added ignore entry for build folder

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ CMakeCache.txt
 *.log
 *.tlog
 build
+build-*/
```
Comment (StevenPuttemans) [inline .gitignore]: And why do we need this?

**Comment by alalek** [comment]
I believe this approach should work for you: http://stackoverflow.com/a/7335487

File: .gitignore (hunk 1 above)
Comment (salimoneus) [inline .gitignore]: The existing "build" entry is not matching the build folder on at least one platform using the latest Git (Windows) thus UI clients show thousands of Untracked entries in the file view.  Adding "ignore = dirty" or "ignore = untracked" to the .gitmodules also does not seem to ignore those files.

Comment (salimoneus) [inline .gitignore]: opencv

opencv2

Comment (alalek) [inline .gitignore]: What build script specifies that build directory name?

**Pull request closed without merging**